src/circuit/gadget/sinsemilla/chip.rs

```diff
@@ -65,8 +65,14 @@ pub struct SinsemillaConfig {
     /// x-coordinate of the domain $Q$, which is then constrained to equal the
     /// initial $x_a$.
     constants: Column<Fixed>,
-    /// Permutation over all advice columns and the `constants` fixed column.
-    perm: Permutation,
+    // Permutation over all advice columns and the `constants` fixed column.
+    pub perm: Permutation,
+}
+
+impl SinsemillaConfig {
+    fn advices(&self) -> [Column<Advice>; 5] {
+        [self.bits, self.x_a, self.x_p, self.lambda_1, self.lambda_2]
```

Comment:
The order of columns here doesn't matter because `advices()` is only called outside the context of the Sinsemilla chip, e.g. for another chip which uses the same number of columns, but in an arbitrary order. If a chip needs to use the exact `SinsemillaConfig`, it should contain a copy of `SinsemillaConfig`:

```rust
MerkleConfig {
    sinsemilla_config: SinsemillaConfig
}
```
In this PR, `advices()` is used to provide 5 advice columns to the `CondSwapChip`, which doesn't care about Sinsemilla.

Reply:
Okay; this is likely somewhere we will fiddle with column orderings then once the full circuit is assembled, so we can position smaller allocations appropriately.